src/client/datascience/notebook/intellisense/pythonKernelCompletionProvider.ts

```diff
@@ -91,13 +91,14 @@ export class PythonKernelCompletionProvider implements CompletionItemProvider {
         // Allow slower timeouts for CI (testing).
         const timeout =
             parseInt(process.env.VSC_JUPYTER_IntellisenseTimeout || '0', 10) || Settings.IntellisenseTimeout;
+        const pylanceTimeout = Math.min(1000, timeout);
```

Comment:
This feels a bit funny combined with a user controlled setting. If I specify 5000 for IntellisenseTimeout I think that I'd expect that to be the timeout, not 1000.